nixos/modules/system/boot/systemd.nix

```
@@ -226,7 +226,24 @@ in
     package = mkPackageOption pkgs "systemd" { };
 
     enableStrictShellChecks = mkEnableOption "" // {
-      description = "Whether to run shellcheck on the generated scripts for systemd units.";
+      description = ''
+        Whether to run shellcheck on the generated scripts for systemd units.
+        When enabled, all systemd scripts generated by NixOS will be checked with
+        shellcheck and any errors or warnings will cause the build to fail. This
+        affects all scripts that have been created through the script, reload,
+        preStart, postStart, preStop and postStop options for systemd services.
+        This does not affect commandlines passed directly to ExecStart,
+        ExecReload, ExecStartPre, ExecStartPost, ExecStop or ExecStopPost. It
+        therefore also does not affect systemd units that are coming from
```

```suggestion
        ExecReload, ExecStartPre, ExecStartPost, ExecStop or ExecStopPost.
        
        Therefore, this does not affect systemd units that are coming from
```
I recommend splitting this into more paragraphs. It's quite hard to read as it is.

---

sorry not used to accepting suggestions and how that works on github, i reimplemented this myself.  I will look at this next round.  let me know if i missed anything, sorry for the inconvenience.

---

GitHub's accept suggestion button usually mangles the commit history, so it's best to not use it in Nixpkgs. Don't forget to squash your commits.